pkg/controllers/disruption/consolidation.go

```diff
@@ -192,6 +193,49 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 		results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
 	}
 
+	results.NewNodeClaims[0].InstanceTypeOptions = instanceTypesOptionsFilteredByPrice
+	return Command{
+		candidates:   candidates,
+		replacements: results.NewNodeClaims,
+	}, nil
+}
+
+// Allow spot consolidation if SpotConsolidation featureflag is set to true by customers:
+// Fetch all the possible InstanceTypes for a candidate from simulation, sort it by price.
+// Only allow spot consolidation if we have atleast 15 cheapest instance types to replace the current candidate.
+// For single node consolidation, check if the current candidate is part of the first 15 cheapest instance types inorder to avoid repeated consolidation. If yes, do not consolidate.
+// If no, consolidate by sending the filtered list of instanceTypes (instanceTypesOptionsFilteredByPrice) for launch because we want to replace the current candidate with the cheaper instanceType.
+func (c *consolidation) checkForSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate,
```

Comment:
```suggestion
func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate,
```

nit: This is less of a "check" (which I would expect to return a bool) and more of a "compute command" operation

Reply:
Addressed